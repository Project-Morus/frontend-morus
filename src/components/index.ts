
// Buttons
export { default as Button } from './Buttons/Button'
export { default as RadioButtonGroup } from './Buttons/RadioButtonGroup'
export { default as RadioButton } from './Buttons/RadioButton'

// DataDisplay
export { default as Icon } from './DataDisplay/Icon'
export { default as IconCard } from './DataDisplay/IconCard'
export { default as SequencialStepper } from './DataDisplay/SequencialStepper'
export { default as Table } from './DataDisplay/Table'

export { default as Archive } from './DataDisplay/Archive'

// Feedback
export { default as Aside } from './Feedback/Aside'
export { default as Divider } from './Feedback/Divider'
export { default as Modal } from './Feedback/Modal'


// Inputs
export { default as Input } from './Inputs/Input'

// Navigations
export { default as Menu } from './Navigation/Menu';
