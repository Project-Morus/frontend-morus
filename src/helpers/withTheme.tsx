import { ComponentType, JSX } from 'react';
import { ThemeProvider } from "styled-components";
import { defaultTheme } from "../styles/themes/default";

type ComponentPropsType<T> = T & JSX.IntrinsicAttributes

export function withTheme<T>(WrapperComponent: ComponentType<ComponentPropsType<T>>) {
  return (props: ComponentPropsType<T>) => {
    return (
      <ThemeProvider theme={defaultTheme}>
        <WrapperComponent {...props} />
      </ThemeProvider>
    );
  };
}

