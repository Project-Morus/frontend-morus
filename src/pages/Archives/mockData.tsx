import { ITableHeaderProps } from "../../components/DataDisplay/Table/types";
import Icons from "./parts/Icons";


export const HEADER_TABLE_CELLS: ITableHeaderProps[] = [
  { colName: "Tamanho" },
  { colName: "Nome" },
  { colName: "Data da postagem" },
  { colName: "Ações" },
];

export const ROW_NAME_CELLS = [
  { size: "35MB", name: "Regimento Interno do Condomínio Ilha de Capri", created_at: "11/05/2023", actions: <Icons /> },
  { size: "15MB", name: "Convenção do condomínio Ilha de Capri", created_at: "12/04/2023", actions: <Icons /> },
  { size: "2MB", name: "Edital de Convocação para Assembleia Geral Ordinária", created_at: "10/04/2023", actions: <Icons /> },
  { size: "7MB", name: "Alvará de construção da prefeitura de Vila Velha-ES", created_at: "12/02/2022", actions: <Icons /> },
  { size: "101MB", name: "Plantas e Projetos", created_at: "19/01/2022", actions: <Icons /> },
];
