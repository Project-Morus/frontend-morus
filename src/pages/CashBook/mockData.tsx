import Icon from "../../components/Icon";
import { ITableHeaderProps } from "../../components/Table/types";

const IconsSC = () => {
  return (
    <div style={{ display: "flex", gap: "0.8rem" }}>
      <Icon name="ph-pencil-simple-line" size="20" />
      <Icon name="ph-trash" color="#CE323B" size="20" />
    </div>
  );
};

export const HEADER_TABLE_CELLS: ITableHeaderProps[] = [
  { colName: "Torre" },
  { colName: "Descrição" },
  { colName: "Valor" },
  { colName: "Data" },
  { colName: "Categoria" },
  { colName: "Ações" },
];

export const ROW_NAME_CELLS = [
  {
    tower: "Todos",
    description: "Salário do porteiro",
    value: "-R$3.250,50",
    date: "05/02/2024",
    category: "Funcionário",
    actions: <IconsSC />,
  },
  {
    tower: "Norte",
    description: "Taxa de condomínio",
    value: "R$1.200,50",
    date: "05/02/2024",
    category: "Taxa de condomínio",
    actions: <IconsSC />,
  },
  {
    tower: "Sul",
    description: "Taxa de condomínio",
    value: "R$1.200,50",
    date: "05/02/2024",
    category: "Taxa de condomínio",
    actions: <IconsSC />,
  },
  {
    tower: "Sul",
    description: "Taxa de condomínio",
    value: "R$1.200,50",
    date: "05/02/2024",
    category: "Taxa de condomínio",
    actions: <IconsSC />,
  },
  {
    tower: "Norte",
    description: "Taxa de condomínio",
    value: "R$1.200,50",
    date: "05/02/2024",
    category: "Taxa de condomínio",
    actions: <IconsSC />,
  },
];
