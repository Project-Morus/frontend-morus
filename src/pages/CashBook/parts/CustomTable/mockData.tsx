import { ITableHeaderProps } from "../../types";

export const HEADER_TABLE_CELLS: ITableHeaderProps[] = [
  { colName: "Torre" },
  { colName: "Descrição" },
  { colName: "Valor" },
  { colName: "Data" },
  { colName: "Categoria" },
  { colName: "Ações" },
];
