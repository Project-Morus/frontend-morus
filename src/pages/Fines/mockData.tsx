import { ITableHeaderProps } from "../../components/DataDisplay/Table/types";

export const HEADER_TABLE_CELLS: ITableHeaderProps[] = [
  { colName: "Nome" },
  { colName: "Motivo" },
  { colName: "Valor" },
  { colName: "Juros" },
  { colName: "Data" },
  { colName: "Vencimento" },
  { colName: "Status" },
  { colName: "Ações" },
];
