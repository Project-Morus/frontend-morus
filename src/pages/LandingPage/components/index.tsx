export * from './InformationSection';
export * from './HeroSection';
export * from './CommonAreaSection';
export * from './CustomersSection';
export * from './FooterSection';