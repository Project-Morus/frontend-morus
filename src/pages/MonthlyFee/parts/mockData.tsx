import { ITableHeaderProps } from "../../../components/DataDisplay/Table/types";

export const HEADER_TABLE_CELLS: ITableHeaderProps[] = [
  { colName: "Nome" },
  { colName: "Descrição" },
  { colName: "Valor" },
  { colName: "Vencimento" },
  { colName: "Status" },
  { colName: "Ações" },
];
