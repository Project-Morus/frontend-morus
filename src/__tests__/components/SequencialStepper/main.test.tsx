import { render, screen } from '@testing-library/react'
import { withTheme } from '../../../helpers/withTheme'
import { SequencialStepper }from '../../../components'
import { SEQUENTIAL_STEPPER_DATA } from './mockData'

const SequencialStepperWithTheme = withTheme(SequencialStepper)

describe('SequencialStepper component', () => {
  test('Verify if component is rendering', () => {
    render(<SequencialStepperWithTheme role='roleSequecialStepper' items={SEQUENTIAL_STEPPER_DATA} currentStep={1} />)

    expect(screen.getByRole('roleSequecialStepper'))
  })

  test('Snapshot SequencialStepper', () => {
    const { asFragment } = render(
      <SequencialStepperWithTheme items={SEQUENTIAL_STEPPER_DATA} currentStep={1} />,
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test('Snapshot SequencialStepper with currentStep in two', () => {
    const { asFragment } = render(
      <SequencialStepperWithTheme items={SEQUENTIAL_STEPPER_DATA} currentStep={2} />,
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test('Snapshot SequencialStepper with currentStep in last step', () => {
    const { asFragment } = render(
      <SequencialStepperWithTheme items={SEQUENTIAL_STEPPER_DATA} currentStep={3} />,
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
