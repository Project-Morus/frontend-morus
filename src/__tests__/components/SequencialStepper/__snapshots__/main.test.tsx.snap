// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SequencialStepper component Snapshot SequencialStepper 1`] = `
<DocumentFragment>
  .c0 {
  display: flex;
  align-items: center;
  flex-direction: row;
  gap: 2.25rem;
}

.c1 {
  width: 15px;
  height: 15px;
  border-radius: 50%;
  background: #3575B6;
}

.c2 {
  width: 15px;
  height: 15px;
  border-radius: 50%;
  background: #B1CAE3;
}

<div
    class="c0"
  >
    <div
      class="c1"
    />
    <div
      class="c2"
    />
    <div
      class="c2"
    />
  </div>
</DocumentFragment>
`;

exports[`SequencialStepper component Snapshot SequencialStepper with currentStep in last step 1`] = `
<DocumentFragment>
  .c0 {
  display: flex;
  align-items: center;
  flex-direction: row;
  gap: 2.25rem;
}

.c1 {
  width: 15px;
  height: 15px;
  border-radius: 50%;
  background: #B1CAE3;
}

.c2 {
  width: 15px;
  height: 15px;
  border-radius: 50%;
  background: #3575B6;
}

<div
    class="c0"
  >
    <div
      class="c1"
    />
    <div
      class="c1"
    />
    <div
      class="c2"
    />
  </div>
</DocumentFragment>
`;

exports[`SequencialStepper component Snapshot SequencialStepper with currentStep in two 1`] = `
<DocumentFragment>
  .c0 {
  display: flex;
  align-items: center;
  flex-direction: row;
  gap: 2.25rem;
}

.c1 {
  width: 15px;
  height: 15px;
  border-radius: 50%;
  background: #B1CAE3;
}

.c2 {
  width: 15px;
  height: 15px;
  border-radius: 50%;
  background: #3575B6;
}

<div
    class="c0"
  >
    <div
      class="c1"
    />
    <div
      class="c2"
    />
    <div
      class="c1"
    />
  </div>
</DocumentFragment>
`;
